imports/client/components/HuntNav.tsx

```diff
@@ -0,0 +1,216 @@
+import { Meteor } from "meteor/meteor";
+import { useTracker } from "meteor/react-meteor-data";
+import { faBullhorn } from "@fortawesome/free-solid-svg-icons/faBullhorn";
+import { faFaucet } from "@fortawesome/free-solid-svg-icons/faFaucet";
+import { faMap } from "@fortawesome/free-solid-svg-icons/faMap";
+import { faReceipt } from "@fortawesome/free-solid-svg-icons/faReceipt";
+import { faUsers } from "@fortawesome/free-solid-svg-icons/faUsers";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import React from "react";
+import Button from "react-bootstrap/Button";
+import { useParams } from "react-router-dom";
+import styled, { css } from "styled-components";
+import Hunts from "../../lib/models/Hunts";
+import { userMayWritePuzzlesForHunt } from "../../lib/permission_stubs";
+import { mediaBreakpointDown } from "./styling/responsive";
+import { Nav } from "react-bootstrap";
+
+const HuntLinkWrapper = styled.div`
+  border-color: #cfcfcf;
+  border-style: solid;
+  border-width: 0 0 0 1px;
+  margin-left: 8px;
+  padding-left: 8px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  max-height: 50px;
+  overflow-y: hidden;
+
+  ${mediaBreakpointDown(
+    "lg",
+    css`
+      border-left-width: 0;
+    `,
+  )}
+
+  ${mediaBreakpointDown(
+    "sm",
+    css`
+      margin-left: 0;
+      padding-left: 0;
+      flex: 1;
+
+      a {
+        border-radius: 0;
+        flex: 1;
+      }
+    `,
+  )}
+`;
+
+const JRLinkList = styled(Nav)`
+  border-color: #cfcfcf;
+  border-style: solid;
+  border-width: 0 1px 0 0;
+  margin-right: 8px;
+  padding-right: 8px;
+  flex: 1 1 auto;
+  display: flex;
+  justify-content: flex-end;
+
+  ${mediaBreakpointDown(
+    "lg",
+    css`
+      border-right-width: 0;
+    `,
+  )}
+
+  ${mediaBreakpointDown(
+    "sm",
+    css`
+      justify-content: space-between;
+      border: 1px solid #0d6efd;
+      margin-right: 0;
+      padding-right: 0;
+    `,
+  )}
+`;
+
+const StyledPuzzleListLinkAnchor = styled(Nav.Link)`
```

Comment:
I'm noticing that after this change, all the nav items are triggering full page reloads (loading html, code, & assets, opening a new websocket, etc.), rather than doing in-app navigation.

I think in the old behavior this was a styled `react-router-dom` `Link`, which overrides some handlers to avoid triggering navigation.  Can you make sure that element is in the component hierarchy, and verify that following these links trigger only in-app navigation, rather than browser-level navigation?

Reply:
update: looks like the tool of choice (since you want to provide the `<a>` element itself) is `react-router-bootstrap`'s `<LinkContainer>`